render.js

```diff
@@ -156,50 +183,83 @@ function screenshots_html(result) {
     }).join('\n');
 }
 
+function _calcSingleStatusStr(status) {
+    return ({
+        'success': '✔️',
+        'error': '✘',
+    }[status] || status);
+}
+
+function _calcSummaryStatus(taskResults) {
+    if (taskResults.every(tr => tr.status === taskResults[0].status)) {
+        return _calcSingleStatusStr(taskResults[0].status);
+    }
+
+    // Flaky results, tabulate
+    const counter = new Map();
+    for (const tr of taskResults) {
+        counter.set(tr.status, (counter.get(tr.status) || 0) + 1);
+    }
+
+    return (Array.from(counter.keys())
+        .sort()
+        .map(status => counter.get(status) + _calcSingleStatusStr(status))
+        .join(' '));
+}
+
+function _calcDuration(taskResults) {
+    if (taskResults.length === 1) {
+        return format_duration(taskResults[0].duration);
+    }
+
+    const max = Math.max(... taskResults.map(tr => tr.duration));
+    const min = Math.min(... taskResults.map(tr => tr.duration));
```

Comment:
Minor nit: We could skip the second iteration by storing the duration array in a variable.

Reply:
Excellent point! Fixed. If only JavaScript had [generator expressions](https://docs.python.org/dev/reference/expressions.html#grammar-token-generator-expression) ;)